lib/smart_answer_flows/benefit-cap-calculator.rb

```diff
@@ -175,5 +175,8 @@ def define
       ## Outcome 5
       outcome :outcome_not_affected
     end
+    def self.next_benefit_amount_question(benefits, selected_benefits)
+      benefits.fetch(selected_benefits.shift, :housing_benefit_amount?)
+    end
```

Comment:
I find it confusing that `.next_benefit_amount_question` modifies the `benefit_types` array and returns the name of the next question. I actually wonder whether it might be better to live with the duplication of `benefits.fetch(benefit_types.shift...` as at least that makes it clearer that `benefit_types` is being modified.